### PR TITLE
Build System Overhaul

### PR DESCRIPTION
**Pull request by Kai-Richardson**
### :page_facing_up: Summary of Changes

This PR replaces a significant portion of our existing build system with dependency-driven [wireit](https://github.com/google/wireit) scripts. Since we're not currently using a monorepo tool, this is a great middle ground.

The existing commands have been simplified, and everything Should Just Work™️.

This PR also introduces something new to Modus WC - hot reloading storybook working with Stencil. This will be an invaluable development tool, as we can now use Storybook to do development on new and existing components, instead of having to scaffold them into a stripped web server.

I aim to make an even more comprehensive and easy-to read README in the future.

Here is a diagram of all available build scripts, also at build-scripts.md:

```mermaid
graph LR
    build --> lint_1
    build --> stencil_build
    build --> storybook_build
    build:ci --> stencil_build
    stencil_build --> core_tasks
    storybook_build --> stencil_build
    storybook_start --> stencil_watch
    stencil_watch --> core_tasks
    format --> format_prettier
    format --> format_eslint
    lint --> lint_prettier
    lint --> lint_eslint
    start --> storybook_start
    start --> lint_2
    clean --> desc0["Clean all build artifacts manually."]
    stencil:generate --> desc1["Starts the interactive generator for a new Stencil component."]
    test --> desc2["Runs spec tests using Stencil & Jest"]
    test:coverage --> desc3["Runs Jest tests and outputs a coverage report."]
    test:update-snapshot --> desc4["Updates Jest snapshots for Stencil tests."]
    test:watch --> desc5["Watches for changes and re-runs Stencil tests."]

    subgraph complex npm run scripts
        build
        build:ci
        format
        lint
        start
    end

    subgraph wireit tasks

        subgraph core_tasks["core tasks"]
            tailwind
            build_cem_json
        end

        lint_1["`
            **lint**
        `"]
        lint_2["`
            **lint**
        `"]
        stencil_build["`
            **stencil:build**
            Builds the Stencil components for production.
        `"]
        storybook_build["`
            **storybook:build**
            Builds the Storybook static site.
        `"]
        build_cem_json["`
            **build:cem-json**
            Generates the custom-elements.json file for Stencil components.
        `"]
        stencil_watch["`
            **stencil:watch**
            Watches for changes in the Stencil components and rebuilds.
        `"]
        storybook_start["`
            **storybook:start**
            Starts Storybook in development mode.
        `"]
        tailwind["`
            **tailwind**
            Generates the Tailwind CSS files.
        `"]
        format_prettier["`
            **format:prettier**
            Formats code using Prettier.
        `"]
        format_eslint["`
            **format:eslint**
            Formats code using ESLint.
        `"]
        lint_prettier["`
            **lint:prettier**
            Lints code using Prettier.
        `"]
        lint_eslint["`
            **lint:eslint**
            Lints code using ESLint.
        `"]
    end

    subgraph simple npm run scripts
        clean
        stencil:generate
        test
        test:coverage
        test:update-snapshot
        test:watch
    end
```

### :thought_balloon: Type of Change

- [x] Bug fix (fixes an issue)
- [x] New feature (adds functionality)
- [x] Configuration change (modifies scaling or properties of existing infrastructure & services)
- [x] Documentation change
- [ ] Other

### :white_check_mark: Self Code Review Checklist

PR authors and reviewers, please verify that all of these items have been completed.

- [x] My code is clean and readable
- [x] I followed standard conventions
- [ ] Component code is WCAG 2.2 compliant (if applicable)
- [ ] I have made theme files changes (if applicable)
- [x] I have made documentation changes (if applicable)
- [x] I have made Storybook changes including usage documentation (if applicable)
- [x] I have tested the component thoroughly in Storybook including RTL rendering (if applicable)

### :link: Azure DevOps Work Item

[AB#571135](https://dev.azure.com/ViewpointVSO/74194628-0b6e-4b03-95a4-2d06c069dcbe/_workitems/edit/571135)

### CODE REVIEW

File: integrations/angular/ng18/projects/trimble-cms/modus-wc-ng/src/lib/stencil-generated/angular-component-lib/utils.ts
Comment (Kai-Richardson) [inline integrations/angular/ng18/projects/trimble-cms/modus-wc-ng/src/lib/stencil-generated/angular-component-lib/utils.ts:1]: These are all stencil-generated files, and shouldn't be checked into Git - they get rebuilt from `npm run build` or etc.

File: src/components/atoms/modus-wc-badge/modus-wc-badge.scss [hunk 1]
```diff
@@ -21,4 +21,4 @@
     color: var(--modus-wc-color-base-content);
     border: none;
   }
-}
\ No newline at end of file
+}
```
Comment (gabriel-piltzer-trimble) [inline src/components/atoms/modus-wc-badge/modus-wc-badge.scss:24]: Did we intentionally remove blank lines at the end of files? It makes it nice when someone adds new lines to a file (similar to hanging commas).

**Review by gabriel-piltzer-trimble — APPROVED** [review]
Very nice work! Left an eslint comment about removing blank lines from end of files.

File: src/components/atoms/modus-wc-badge/modus-wc-badge.scss (hunk 1 above)
Comment (Kai-Richardson) [reply src/components/atoms/modus-wc-badge/modus-wc-badge.scss:24]: This was done on behalf of prettier. We should change the prettier config if we wish to change that behavior. These files were just not properly configured to be picked up previously.

Comment (Kai-Richardson) [reply src/components/atoms/modus-wc-badge/modus-wc-badge.scss:24]: https://prettier.io/docs/en/rationale.html#empty-lines

**Pull request merged**